c10/util/Half.h

```diff
@@ -345,13 +345,9 @@ inline uint16_t fp16_ieee_from_fp32_value(float f) {
   return static_cast<uint16_t>(
       (sign >> 16) |
       (shl1_w > UINT32_C(0xFF000000) ? UINT16_C(0x7E00) : nonsign));
-#endif // X86_F16
+#endif // C10_X86_F16
 }
 
-#ifdef X86_F16
-#undef X86_F16
-#endif // X86_F16
```

Comment:
Why don't you want to undefine it at the end of the file?

Reply:
I thought we were allowed to define macros as long as they were properly namespaced. Seemed unlikely to collide

Reply:
Macros ignore C++ namespaces...

Reply:
> Macros ignore C++ namespaces...

"C10_" is the namespace in my book.

Reply:
anyway I updated the diff to undefine the macro